plugins/woocommerce/tests/e2e-pw/utils/api.js

```diff
@@ -13,6 +13,42 @@ if ( process.env.CONSUMER_KEY && process.env.CONSUMER_SECRET ) {
 	} );
 }
 
+/**
+ * Allow explicit construction of api client.
+ */
+const constructWith = ( consumerKey, consumerSecret ) => {
+	api = new wcApi( {
+		url: config.use.baseURL,
+		consumerKey,
+		consumerSecret,
+		version: 'wc/v3',
+	} );
+};
+
+const throwCustomError = (
+	error,
+	customMessage = 'Something went wrong. See details below.'
+) => {
+	throw new Error(
+		customMessage
+			.concat(
+				`\nResponse status: ${ error.response.status } ${ error.response.statusText }`
+			)
+			.concat(
+				`\nResponse headers:\n${ JSON.stringify(
+					error.response.headers,
+					null,
+					2
+				) }`
+			).concat( `\nResponse data:\n${ JSON.stringify(
+			error.response.data,
+			null,
+			2
+		) }
+` )
+	);
+};
+
```

Comment:
The `@woocommerce/woocommerce-rest-api` library uses `axios`, which in turn does not give a detailed error message when the API call errors out. This function will catch the error, print out the error's status, headers, and body. I personally found it helpful when debugging our E2E tests in this PR so I propose we add this functionality in.